internal/service/dataexchange/event_action_test.go

```diff
@@ -0,0 +1,671 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package dataexchange_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+	"os"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/YakDriver/regexache"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/dataexchange"
+	awstypes "github.com/aws/aws-sdk-go-v2/service/dataexchange/types"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+	"github.com/hashicorp/terraform-provider-aws/internal/acctest"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	"github.com/hashicorp/terraform-provider-aws/internal/create"
+	"github.com/hashicorp/terraform-provider-aws/internal/errs"
+	"github.com/hashicorp/terraform-provider-aws/names"
+
+	tfdataexchange "github.com/hashicorp/terraform-provider-aws/internal/service/dataexchange"
+)
+
+func TestAccDataExchangeEventAction_basic(t *testing.T) {
+	ctx := acctest.Context(t)
+	if testing.Short() {
+		t.Skip("skipping long-running test in short mode")
+	}
+
+	var eventaction dataexchange.GetEventActionOutput
+	resourceName := "aws_dataexchange_event_action.test"
+
+	bucketName := strconv.Itoa(int(time.Now().UnixNano()))
+
+	if _, okAcc := os.LookupEnv("TF_ACC"); !okAcc {
+		t.Skipf("TF_ACC must be set")
+	}
+
+	cfg, err := config.LoadDefaultConfig(ctx)
+	if err != nil {
+		t.Error(err)
+	}
+	stsConn := sts.NewFromConfig(cfg)
+	identity, err := stsConn.GetCallerIdentity(ctx, &sts.GetCallerIdentityInput{})
+
+	dataSetId, err := helperAccEventActionGetReceivedDataSet(ctx, awstypes.AssetTypeS3Snapshot)
+	if err != nil {
+		t.Error(err)
+	}
+
+	if dataSetId == "" {
+		return
+	}
```

Comment:
I'm having trouble identifying the purpose of this setup - is there a way this can be incorporated into the test configuration itself, rather than manual steps outside the scope of the test run?